packages/core/src/external/carequality/iti55/process-incoming-iti-55.ts

```diff
@@ -0,0 +1,202 @@
+  patient: PatientDataMPI
+): PatientDiscoveryResponseOutgoing {
+  return {
+    id: payload.id,
+    timestamp: payload.timestamp,
+    responseTimestamp: new Date().toISOString(),
+    patientMatch: true,
+    xcpdPatientId: {
+      id: patient.id,
+      system: "000", // TBD	- what is this
+    },
+    patientResource: convertPatientToFHIR(patient),
+    xcpdHomeCommunityId: METRIPORT_HOME_COMMUNITY_ID,
+  };
+}
+
+export async function processIncomingRequest(
+  payload: PatientDiscoveryRequestIncoming
+): Promise<PatientDiscoveryResponseOutgoing> {
+  const apiClient = apiClientMPIBlockEndpoint();
+
+  try {
+    const patient = validateFHIRAndExtractPatient(payload.patientResource);
+    const normalizedPatientDemo = normalizePatientDataMPI(patient);
+
+    if (!normalizedPatientDemo) {
+      return constructErrorResponse(
+        payload,
+        "1.3.6.1.4.1.19376.1.2.27.3",
+        "Internal Server Error",
+        "Invalid Patient Data"
+      );
+    }
+
+    const response = await apiClient.callInternalEndpoint({
+      dob: normalizedPatientDemo.dob,
+      genderAtBirth: normalizedPatientDemo.genderAtBirth,
+    });
+    const blockedPatients: PatientDataMPI[] = response.data;
+
+    const matchingPatients = matchPatients(
+      jaroWinklerSimilarity,
+      [matchingPersonalIdentifiersRule],
+      blockedPatients,
+      normalizedPatientDemo,
+      SIMILARITY_THRESHOLD
+    );
+
+    const mpiPatient = await mergePatients(
+      mergeWithFirstPatient,
+      matchingPatients,
+      normalizedPatientDemo
+    );
+
+    if (!mpiPatient) {
+      return constructNoMatchResponse(payload);
+    }
+    return constructMatchResponse(payload, mpiPatient);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    switch (error.constructor) {
+      case InternalError:
+        return constructErrorResponse(
+          payload,
+          "1.3.6.1.4.1.19376.1.2.27.3",
+          "InternalError",
+          error.message
+        );
+      case PatientAddressRequestedError:
+        return constructErrorResponse(
+          payload,
+          "1.3.6.1.4.1.19376.1.2.27.1",
+          "PatientAddressRequested",
+          error.message
+        );
+      case LivingSubjectAdministrativeGenderRequestedError:
+        return constructErrorResponse(
+          payload,
+          "1.3.6.1.4.1.19376.1.2.27.1",
+          "LivingSubjectAdministrativeGenderRequested",
+          error.message
+        );
+      default:
+        return constructErrorResponse(
+          payload,
+          "1.3.6.1.4.1.19376.1.2.27.3",
+          "Internal Server Error",
+          "Unknown Error: Contact Metriport Support for assistance"
+        );
+    }
+  }
+}
+
+export function apiClientMPIBlockEndpoint() {
+  const apiURL = getEnvVarOrFail("API_URL");
+  const postEndpointUrl = `${apiURL}/internal/patient/mpi/block`;
+
+  return {
+    callInternalEndpoint: async function (
```

Comment:
Id call this something else thats a bit more descriptive

Reply:
You probably dont even need to return a function you can just do the actual try catch within this function. That way there is no need to do `const apiClient = apiClientMPIBlockEndpoint();`